ledger/src/shredder.rs

```diff
@@ -83,7 +83,7 @@ impl Shredder {
             keypair,
             &entries,
             is_last_in_slot,
-            chained_merkle_root,
+            Some(chained_merkle_root),
```

Comment:
I'm inclined to say let's make `Shredder: make_shreds_from_data_slice()` also take a `Hash` instead of `Option<Hash>`; there are only a couple more instances and it will make all of the `Shredder` functions consistent.

This seems like a decent line to draw, and can then update the underlying function (below) in a subsequent PR:
https://github.com/anza-xyz/agave/blob/f9294b161b5f1b238542cc751e07ef00c8656708/ledger/src/shred/merkle.rs#L1021-L1026